source/api/AST/ASTVisitor.cpp

```diff
@@ -283,6 +284,67 @@ parseParameters(
     }
 }
 
+TemplateParamInfo
+buildTemplateParam(
+    const NamedDecl* ND)
+{
+    std::string name = ND->getNameAsString();
+    // KRYSTIAN NOTE: Decl::isParameterPack  
+    // returns true for function parameter packs
+    bool is_pack = ND->isTemplateParameterPack();
+    if(const auto* TP = dyn_cast<
+        TemplateTypeParmDecl>(ND))
```

Comment:
Instead of dyn_cast you could add an overload for `TemplateTypeParmDecl`

Reply:
`NamedDecl` to `TemplateTypeParmDecl` is an upcast, so `dyn_cast` is needed. I ended up cleaning this section up a bit so take a look at it now @vinniefalco